File: JWP_API/JadeWesserPort/DTOs/DailyScheduleDTOs/DailyScheduleItemDto.cs
using JadeWesserPort.Domain.Entities;

namespace JadeWesserPort.DTOs.DailyScheduleDTOs;

public class DailyScheduleItemDto
{
    public int VVNCode { get; set; }
    public string DockCode { get; set; } = null!;
    public string[] CraneCodes { get; set; } = [];
    public int[] StaffCodes { get; set; } = [];
    public int Start { get; set; }
    public int End { get; set; }
    public string StorageAreaCode { get; set; } = default!;
    public List<PlannedOperation>? PlannedOperations { get; set; }
}

File: Frontend/src/infrastructure/dtos/scheduling/DailyScheduleItemDto.ts
import type { PlannedOperation } from "../../../domain/Types/entities/PlannedOperation";

export type DailyScheduleItemDto = {
  vvnCode: number;
  dockCode: string;
  craneCodes: string[];
  staffCodes: number[];
  storageAreaCode: string;
  start: number;
  end: number;
  plannedOperations: PlannedOperation[];
};
